tests/test_java_violations_reporter.py

```diff
@@ -417,41 +406,35 @@ def test_quality_pregenerated_report(self):
         # We're not guaranteed that the violations are returned
         # in any particular order.
         actual_violations = quality.violations("path/to/file.java")
-        self.assertEqual(len(actual_violations), len(expected_violations))
+        assert len(actual_violations) == len(expected_violations)
         for expected in expected_violations:
-            self.assertIn(expected, actual_violations)
+            assert expected in actual_violations
 
 
-class PmdXmlQualityReporterTest(unittest.TestCase):
-    def setUp(self):
-        _patch_so_all_files_exist()
+class TestPmdXmlQualityReporterTest:
+    @pytest.fixture(autouse=True)
+    def setup(self, mocker):
         # Paths generated by git_path are always the given argument
-        _git_path_mock = patch(
+        _git_path_mock = mocker.patch(
             "diff_cover.violationsreporters.java_violations_reporter.GitPathTool"
-        ).start()
+        )
         _git_path_mock.relative_path = lambda path: path
         _git_path_mock.absolute_path = lambda path: path
 
-    def tearDown(self):
-        """
-        Undo all patches.
-        """
-        patch.stopall()
-
     def test_no_such_file(self):
         quality = QualityReporter(PmdXmlDriver())
 
         # Expect that we get no results
         result = quality.violations("")
-        self.assertEqual(result, [])
+        assert not result
 
     def test_no_java_file(self):
         quality = QualityReporter(PmdXmlDriver())
         file_paths = ["file1.coffee", "subdir/file2.js"]
         # Expect that we get no results because no Java files
         for path in file_paths:
             result = quality.violations(path)
-            self.assertEqual(result, [])
+            assert not result
```

Comment:
last one (Now that I got here I feel like I should have just said 'replace all of them please')

Reply:
probably, but that made it easy for me :-D